### ISSUE TITLE
KeyError: b'1' in read_string_from_stream

### ISSUE THREAD

**Issue by DL6ER** [py-pdf/PyPDF2#1294]
See https://github.com/py-pdf/PyPDF2/issues/1269 for further details, this reports another issue I've come accross.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Linux-5.4.0-122-generic-x86_64-with-glibc2.29

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
2.10.3
```
## Code + PDF

This is a minimal, complete example that shows the issue:

```python
import PyPDF2
with open("AN EXACT ANALYTICAL SOLUTION OF KEPLER'S EQUATION.pdf", "rb") as f:
  pdfreader = PyPDF2.PdfFileReader(f, strict=False)
  full_content = " ".join([page.extractText() for page in pdfreader.pages])
```

PDF used above: [AN EXACT ANALYTICAL SOLUTION OF KEPLER'S EQUATION.pdf](https://github.com/timedegree/playground_files/blob/1db71cc545a8f487f0257145438cc016523b46c4/%E8%AE%BA%E6%96%87/AN%20EXACT%20ANALYTICAL%20SOLUTION%20OF%20KEPLER%27S%20EQUATION.pdf)

Another example triggering the exact same issue: [How to time-stamp a digital document-88624977e595442fde2b0bd4fe3a0fae.pdf](https://github.com/behfarkhosravi/behfarkhosravi.github.io/blob/391b879169d21716636158857751379af093eb7c/assets/files/How%20to%20time-stamp%20a%20digital%20document-88624977e595442fde2b0bd4fe3a0fae.pdf)

## Traceback

This is the complete Traceback I see:

``` plain
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/generic/_utils.py", line 80, in read_string_from_stream
    tok = escape_dict[tok]
KeyError: b'1'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/_page.py", line 1538, in extractText
    return self.extract_text()
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/_page.py", line 1510, in extract_text
    return self._extract_text(
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/_page.py", line 1157, in _extract_text
    content = ContentStream(content, pdf, "bytes")
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/generic/_data_structures.py", line 689, in __init__
    self.__parse_content_stream(stream_bytes)
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/generic/_data_structures.py", line 719, in __parse_content_stream
    operands.append(read_object(stream, None, self.forced_encoding))
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/generic/_data_structures.py", line 830, in read_object
    return read_string_from_stream(stream, forced_encoding)
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/generic/_utils.py", line 95, in read_string_from_stream
    tok = b_(chr(int(tok, base=8)))
ValueError: invalid literal for int() with base 8: b'169'
```

**The PDF can be read using a normal PDF viewer.**

**Comment by DL6ER** [comment]
Further examples:

`b'128'`:
- [[弦论.第2卷.超弦理论及其超越].String.Theory.II.Superstring.Theory.and.Beyond.Joseph.Polchinski.pdf](https://github.com/timedegree/playground_files/blob/3e5ec19bc705a41df4c63dd22e8ba173f178b588/%E5%85%B6%E4%BB%96%E5%AD%A6%E7%A7%91%E8%B5%84%E6%96%99/%5B%E5%BC%A6%E8%AE%BA.%E7%AC%AC2%E5%8D%B7.%E8%B6%85%E5%BC%A6%E7%90%86%E8%AE%BA%E5%8F%8A%E5%85%B6%E8%B6%85%E8%B6%8A%5D.String.Theory.II.Superstring.Theory.and.Beyond.Joseph.Polchinski.pdf)

`b'148'`:
- [BOOKS_NEW_-_BooksNew_-_med_-_The Human Body - An Introduction to Structure and Function - A. Faller, M. Schuenke (Thieme, 2004) WW.pdf](https://github.com/Lexicanium/Lexicanium/blob/9eba11bef30abeeff56e5940caf756c52319307a/data/named/100/BOOKS_NEW_-_BooksNew_-_med_-_The%20Human%20Body%20-%20An%20Introduction%20to%20Structure%20and%20Function%20-%20A.%20Faller%2C%20M.%20Schuenke%20%28Thieme%2C%202004%29%20WW.pdf)


`b'149'`
- [BOOKS_NEW_-_BooksNew_-_med_-_The Psychology of Stalking - Clinical and Forensic Perspectives - J. Meloy (AP, 1998) WW.pdf](https://github.com/Lexicanium/Lexicanium/blob/0b0bfabbe03011944a114e5993afafecfe9d8c6b/data/named/100/BOOKS_NEW_-_BooksNew_-_med_-_The%20Psychology%20of%20Stalking%20-%20Clinical%20and%20Forensic%20Perspectives%20-%20J.%20Meloy%20%28AP%2C%201998%29%20WW.pdf)

**Comment by pubpub-zz** [comment]
there is no separator between the octal code and the decimal character following: the criteria used (`isdigit`) is not restrictive enough. test reviewed to only accept octal numbers
PR in progress

**Comment by pubpub-zz** [comment]
the PR was forgotten for a while. changed merged in #1297
